lib/basedriver/commands/find.js

```diff
@@ -357,45 +363,122 @@ helpers.getScreenshotForImageFind = async function (screenWidth, screenHeight) {
              `screen at ${screenWidth}x${screenHeight}`);
     imgObj = imgObj.resize(screenWidth, screenHeight);
 
-    scale = {xScale: (1.0 * screenWidth) / shotWidth, yScale: (1.0 * screenHeight) / shotHeight};
+    scale.xScale *= (1.0 * screenWidth) / shotWidth;
+    scale.yScale *= (1.0 * screenHeight) / shotHeight;
   }
 
   b64Screenshot = (await imgObj.getBuffer(imageUtil.MIME_PNG)).toString('base64');
   return {b64Screenshot, scale};
 };
 
+/**
+ * @typedef {Object} ImageTemplateSettings
+ * @property {boolean} fixImageTemplateScale - fixImageTemplateScale in device-settings
+ * @property {float} defaultImageTemplateScale - defaultImageTemplateScale in device-settings
+ * @property {boolean} ignoreDefaultImageTemplateScale - Ignore defaultImageTemplateScale if it has true.
+ * If b64Template has been scaled to defaultImageTemplateScale or should ignore the scale,
+ * this parameter should be true. e.g. click in image-element module
+ * @property {float} xScale - Scale ratio for width
+ * @property {float} yScale - Scale ratio for height
+
+ */
 /**
  * Get a image that will be used for template maching.
  * Returns scaled image if scale ratio is provided.
  *
+ *
  * @param {string} b64Template - base64-encoded image used as a template to be
  * matched in the screenshot
- * @param {ScreenshotScale} scale - scale of screen. Default to `{}`
+ * @param {ImageTemplateSettings} opts - Image template scale related options
  *
  * @returns {string} base64-encoded scaled template screenshot
  */
-helpers.fixImageTemplateScale = async function (b64Template, scale = {}) {
-  if (!scale) {
+const DEFAULT_FIX_IMAGE_TEMPLATE_SCALE = 1;
+helpers.fixImageTemplateScale = async function (b64Template, opts = {}) {
+  if (!opts) {
     return b64Template;
   }
 
-  const {xScale, yScale} = scale;
+  let {
+    fixImageTemplateScale = false,
+    defaultImageTemplateScale = DEFAULT_TEMPLATE_IMAGE_SCALE,
+    ignoreDefaultImageTemplateScale = false,
+    xScale = DEFAULT_FIX_IMAGE_TEMPLATE_SCALE,
+    yScale = DEFAULT_FIX_IMAGE_TEMPLATE_SCALE
+  } = opts;
+
+  if (ignoreDefaultImageTemplateScale) {
+    defaultImageTemplateScale = DEFAULT_TEMPLATE_IMAGE_SCALE;
+  }
+
+  // Default
+  if (defaultImageTemplateScale === DEFAULT_TEMPLATE_IMAGE_SCALE && !fixImageTemplateScale) {
+    return b64Template;
+  }
+
+  // Calculate xScale and yScale Appium should scale
+  if (fixImageTemplateScale) {
+    xScale *= defaultImageTemplateScale;
+    yScale *= defaultImageTemplateScale;
+  } else {
+    // `1 *` makes NaN if defaultImageTemplateScale is not a number
+    xScale = yScale = 1 * defaultImageTemplateScale;
+  }
+
+  // xScale and yScale can be NaN if defaultImageTemplateScale is string, for example
   if (!xScale || !yScale) {
     return b64Template;
   }
 
+  // Return if the scale is default (1.0) value
+  if ((xScale - DEFAULT_FIX_IMAGE_TEMPLATE_SCALE) <= 0 && (yScale - DEFAULT_FIX_IMAGE_TEMPLATE_SCALE) === 0) {
```

Comment:
please avoid comparing float values for equality.
It is mandatory to have some delta epsilon value or use toFixed transformation and compare these are strings

Reply:
https://stackoverflow.com/questions/3343623/javascript-comparing-two-float-values